migrator/init.lua

```diff
@@ -35,77 +35,80 @@ end
-            log.error('Errors happened during migrations, gods help you now: %s', json.encode(errors))
-            error(errors[1])
-        end
+            local errors = {}
+            for _, f in pairs(fibers) do
+                local ok, join_result = f:join()
+                if not ok then table.insert(errors, join_result) end
+            end
+            if #errors > 0 then
+                log.error('Errors happened during migrations, gods help you now: %s', json.encode(errors))
+                error(errors[1])
+            end
 
-        log.verbose('All fibers joined, results are: %s', json.encode(result))
+            log.verbose('All fibers joined, results are: %s', json.encode(result))
 
-        for instance_uri, applied in pairs(result) do
-            if not utils.compare(applied, target_names) then
-                error('Not all migrations applied on %s. Actual list: %s', instance_uri, json.encode(applied))
+            for instance_uri, applied in pairs(result) do
+                if not utils.compare(applied, target_names) then
+                    error('Not all migrations applied on %s. Actual list: %s', instance_uri, json.encode(applied))
+                end
             end
-        end
 
-        local config = confapplier.get_deepcopy('migrations') or {}
-        config.applied = config.applied or {}
-        for _, name in ipairs(target_names) do
-            table.insert(config.applied, name)
-        end
+            local config = confapplier.get_deepcopy('migrations') or {}
+            config.applied = config.applied or {}
+            for _, name in ipairs(target_names) do
+                table.insert(config.applied, name)
+            end
 
-        local patch = {
-            migrations = config,
-            ['schema.yml'] = vars.use_cartridge_ddl and ddl.get_schema() or nil
-        }
-        log.verbose('All migrations applied successfully, changing cluster-wide configuration with a patch: %s', json.encode(patch))
+            local patch = {
+                migrations = config,
+                ['schema.yml'] = vars.use_cartridge_ddl and ddl.get_schema() or nil
+            }
+            log.verbose('All migrations applied successfully, changing cluster-wide configuration with a patch: %s', json.encode(patch))
 
-        local _, err = cartridge.config_patch_clusterwide(patch)
-        if err ~= nil then error(err) end
-        log.info('Migrations applied successfully!')
+            local _, err = cartridge.config_patch_clusterwide(patch)
+            if err ~= nil then error(err) end
+            log.info('Migrations applied successfully!')
 
-        local resp = req:render({ json = { applied = target_names }})
-        resp.status = 200
-        return resp
-    end)
+            local resp = req:render({ json = { applied = target_names }})
+            resp.status = 200
+            return resp
+        end)
+        log.info('/migrations/up handler registered successfully')
+    end
```

Comment:
I suggest writing a warning to the logs that http interface is disabled.

Reply:
I desagree. Why we have to warn about impossibility to register route on http server that was disabled?

Reply:
Closing this conversation